components/Themes/src/MDCTonalPalette.m

```diff
@@ -0,0 +1,47 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCTonalPalette.h"
+
+@implementation MDCTonalPalette
+
+- (nonnull instancetype)initWithColors:(nonnull NSArray<UIColor *> *)colors
+                        mainColorIndex:(int)mainColorIndex
+                       lightColorIndex:(int)lightColorIndex
+                        darkColorIndex:(int)darkColorIndex {
+  self = [super init];
+  if (self) {
+    _colors = colors;
+    _mainColorIndex = mainColorIndex;
+    _lightColorIndex = lightColorIndex;
+    _darkColorIndex = darkColorIndex;
```

Comment:
If any of these indexes are out of bounds should we assert, throw and exception, return nil?